app/models/user.rb

```diff
@@ -1,11 +1,25 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+  devise :database_authenticatable,
+         :registerable,
+         :recoverable,
+         :rememberable,
+         :validatable,
+         :omniauthable,
+         omniauth_providers: [:openid_connect]
 
   has_many :memberships
   has_many :groups, through: :memberships
   has_many :lendings
   has_many :reservations
+
+  def self.from_omniauth(auth)
+    # Check if user with provider ('openid_connect') and uid is in db, otherwise create it
```

Comment:
this comment restates the code below, I'm not sure if it helps a lot. I'd like a comment about the use of `from_omniauth`, though - from what i'm seeing, this routine is responsible for syncing the information we get from OIDC into our database, maybe we could put that into a method comment.